File: 2016/src/day05/task1/Task1.java
package day05.task1;

import java.math.BigInteger;
import java.nio.file.Files;
import java.nio.file.Path;
import java.security.MessageDigest;
import java.time.Duration;
import java.time.Instant;

public class Task1 {
   Path path = Path.of("inputs", "day05", "input.txt");

   void task() {
      try {
         var doorId = Files.readString(this.path).trim();
         var result = new StringBuilder();
         var md5 = MessageDigest.getInstance("MD5");

         var suffix = 0;
         while (result.length() < 8) {
            suffix++;
            var hash = new BigInteger(1, md5.digest((doorId + suffix).getBytes())).toString(16);
            while (hash.length() < 32) {
               hash = "0" + hash;
            }
            if (hash.startsWith("00000")) {
               result.append(hash.charAt(5));
            }
         }
         System.out.println(result);
      } catch (Exception e) {
         e.printStackTrace();
         System.exit(1);
      }
   }

   public static void main(String[] args) {
      var start = Instant.now();

      new Task1().task();

      System.out.println("Done in " + Duration.between(start, Instant.now()).toString().substring(2).toLowerCase());
   }
}

File: 2017/src/day09/task1.ts
import path from "node:path";

const PATH = path.join("inputs", "day09", "input.txt");

async function task1() {
   const input = (await Bun.file(PATH).text()).trim();

   let result = 0;
   let level = 0;
   let isGarbage = false;
   for (let i = 0; i < input.length; i++) {
      const character = input.charAt(i);

      if (character == "!") {
         i += 1;
         continue;
      }
      if (isGarbage) {
         if (character == ">") {
            isGarbage = false;
         }
         continue;
      }

      switch (character) {
         case "{":
            level += 1;
            break;
         case "}":
            result += level;
            level -= 1;
            break;
         case "<":
            isGarbage = true;
            break;
      }
   }

   console.log(result);
}

console.time("Task Done");
await task1();
console.timeEnd("Task Done");
